libnavigation-ui/src/main/java/com/mapbox/navigation/ui/voice/SpeechPlayerProvider.java

```diff
@@ -114,9 +114,18 @@ void onDestroy() {
   }
 
   void onSpeechPlayerStateChanged(@NonNull SpeechPlayerState speechPlayerState) {
-    this.speechPlayerState = speechPlayerState;
+    if (speechPlayerState == SpeechPlayerState.OFFLINE_PLAYER_INITIALIZED) {
+      if (this.speechPlayerState == SpeechPlayerState.OFFLINE_PLAYER_INITIALIZING) {
+        this.speechPlayerState = SpeechPlayerState.IDLE;
+      } else {
+        return;
+      }
+    } else {
+      this.speechPlayerState = speechPlayerState;
+    }
```

Comment:
Wondering if we could simplify this confusing `if else` logic.

Reply:
I agree this logic is confusing. The idea is the `OFFLINE_PLAYER_INITIALIZED` will only be used to update the state to `IDLE` when the state is still the initial one, `OFFLINE_PLAYER_INITIALIZING`. Because when the connection is good, we will be playing the first voice via Polly so `state -> ONLINE_PLAYING` and then TTS initialized successfully, we don't want to update the state in this case.

But I think we can remove the `else return` block.

Reply:
> But I think we can remove the else return block.

Yeah, why wouldn't we want to get a notification in such a case? 👍